src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2162,28 +2165,23 @@ LogicalResult ONNXConcatOp::inferShapes() {
   // Negative axis means values are counted from the opposite side.
   if (axisIndex < 0) {
     axisIndex = commonRank + axisIndex;
+    // Tong Chen:
+    // TOFIX: attribute modification should be into canonicalization
```

Comment:
Oh, I like moving such changes to canonicalization... probably could do a sweep for other things as well. I would like to do the same for other constants, including tensors of constants... but I did not really know how to update them. Would like to know how if you do.

Reply:
We will clean up our code later.